src/Fantasma/Internal/Providers/Sql/SqlCluster.cs

```diff
@@ -0,0 +1,151 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Fantasma.Internal;
+
+internal sealed class SqlCluster : ICluster
+{
+    private readonly SqlHeartbeatThread _heartbeat;
+    private readonly TimeProvider _time;
+    private readonly ILogger<SqlCluster> _logger;
+    private readonly IServiceScope _scope;
+    private readonly IFantasmaDatabase _database;
+
+    public bool IsLeader { get; private set; }
+    public bool HasPeers { get; private set; }
+
+    public SqlCluster(
+        SqlHeartbeatThread heartbeat,
+        TimeProvider time,
+        ILogger<SqlCluster> logger,
+        IServiceProvider services)
+    {
+        ArgumentNullException.ThrowIfNull(services);
+
+        _heartbeat = heartbeat ?? throw new ArgumentNullException(nameof(heartbeat));
+        _time = time ?? throw new ArgumentNullException(nameof(time));
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        _scope = services.CreateScope();
+        _database = _scope.ServiceProvider.GetRequiredService<IFantasmaDatabase>();
+    }
+
+    public void Dispose()
+    {
+        _database.Dispose();
```

Comment:
I'm not sure if this can have potential issues with EF Core database pooling, it may need an investigation to make sure this doesn't break it

Reply:
I believe a scoped dependency is disposed of normally, so it shouldn't be a problem. I'm happy to be wrong, though!